### ISSUE TITLE
[FreeBSD 14] Build breaks on arm64: ld: error: undefined symbol: brk

### ISSUE THREAD

**Issue by yurivict** [gperftools/gperftools#1499]
```
ld: error: undefined symbol: brk

referenced by mmap_hook.cc
              src/low_level_alloc_unittest-mmap_hook.o:(sbrk)
did you mean: sbrk
defined in: src/low_level_alloc_unittest-mmap_hook.o
```

[brk isn't supposed to be used in general.](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=278009#c1)

Version: 2.15
FreeBSD 14.0

**Comment by alk** [comment]
Thanks for bug report. So looking closer, looks like our mmap hooks simply assumes that FreeBSD can do sbrk and looks like perhaps 14.0 doesn't anymore. I.e. without even looking at config.h where we have indication of sbrk availability. I welcome patch to fix this.

As for more general matter of not using sbrk. So, yes, in principle we should consider. But in practical terms, just doing our current version of mmap allocator will actually break large heap fragmentation unittest. Yes, this test's specific case is quite subtle, but it is one way to show that sbrk's built-in feature of allocating memory sequentially does give us substantial benefit. There is also potential for another benefit (and potential security downside) which is that brk area being immediately after bss makes certain TLB-miss effects cheaper. I.e. heap sitting closer (likely same pmd) to .bss/.data has possibly measurable performance benefit by having shorter page table walks.

So, yes, I will accept patches and benchmarks towards doing mmap-only allocation, but we cannot simply amputate sbrk just yet.

**Comment by zxombie** [comment]
sbrk and brk have intentionally never been in a release of FreeBSD on arm64 since first released in 11.0 in 2016.
